src/tyr/serializers.cc

```diff
@@ -106,10 +120,14 @@ json::ArrayPtr waypoints(const valhalla::Trip& trip) {
   const auto& legs = trip.routes(0).legs();
   for (const auto& leg : legs) {
     for (const auto& location : leg.location()) {
-      if (&location == &leg.location(0) && &leg != &*legs.begin()) {
+      const auto& is_first_location = &location == &leg.location(0);
+      if (is_first_location && &leg != &*legs.begin()) {
+        // Skip first waypoint of all but first leg so we don't repeat
         continue;
       }
-      waypoints->emplace_back(waypoint(location, false));
+      const auto& edge =
+          is_first_location ? &leg.node(0).edge() : &leg.node(leg.node_size() - 2).edge();
+      waypoints->emplace_back(waypoint(location, *edge, false, false));
```

Comment:
why does `edge` need to be dereferenced? the previous line has it as a const reference.